app/code/Magento/Swatches/view/frontend/templates/product/view/renderer/attributes.phtml

```diff
@@ -0,0 +1,103 @@
+    >
+        <span
+            id="<?= $escaper->escapeHtml($attributeLabelId) ?>"
+            class="<?= $escaper->escapeHtml($attributeLabelClass) ?>"
+        ><?= $escaper->escapeHtml($attributeLabel) ?></span>
+        <span class="<?= $escaper->escapeHtml($attributeSelectedClass) ?>"></span>
+        <div
+            aria-activedescendant=""
+            tabindex="0"
+            aria-invalid="false"
+            aria-required="true"
+            role="listbox"
+            aria-labelledby="<?= $escaper->escapeHtml($attributeLabelId) ?>"
+            class="<?= $escaper->escapeHtml($classes['attribute_options_wrapper']) ?>"
+        >
+            <?php
+            if (isset($swatchConfig[$attributeId])) {
+                $attributeConfig = $swatchConfig[$attributeId];
+                $displayedOptions = 0;
+
+                $swatchOption = $block
+                    ->getChildBlock('swatch_option')
+                    ->setSizeConfig($sizeConfig)
+                    ->setAttributeLabelId($attributeLabelId)
+                    ->setClasses($classes);
+
+                foreach ($attribute['options'] as $optionIndex => $option) {
+                    $optionId = $option['id'];
+                    $optionConfig = $attributeConfig[$optionId] ?? null;
+
+                    if (!$optionConfig) {
+                        continue;
+                    }
+
+                    if ($block->getMoreButtonLimit() === $displayedOptions++) {
+                        $moreButtonClass = $classes['more_button'];
+                        $moreButtonText = $block->getMoreButtonText();
+                        echo '<a href="#" class="' . $escaper->escapeHtml($moreButtonClass) .'"><span>' .
+                            $escaper->escapeHtml($moreButtonText) . '</span></a>';
+                    }
+
+                    echo $swatchOption->setOption($option)
+                        ->setOptionConfig($optionConfig)
+                        ->setIndex($optionIndex)
+                        ->toHtml();
+                }
+            } else {
+                if (!$block->getOnlySwatches()) {
+                    echo $block->getChildBlock('swatch_select')
+                        ->setData('attribute', $attribute)
+                        ->setChooseText($block->getChooseText())
+                        ->setClasses($classes)
+                        ->toHtml();
+                }
+            }
+            ?>
+        </div>
+        <input
+            class="<?= $escaper->escapeHtml($classes['attribute_input']) ?> super-attribute-select"
+            name="super_attribute[<?= $escaper->escapeHtml($attributeId) ?>]"
+            value=""
+            type="text"
+            data-selector="super_attribute[<?= $escaper->escapeHtml($attributeId) ?>]"
+            data-validate="{required: true, messages: {required: '<?= __('Please select %1', $escaper->escapeHtml($attributeLabel)) ?>'}}"
```

Comment:
It's better to escape entire phrase, as in "please select" might be translated to `'"><script>...my script here</script><input type="hidden" data-validate="...'` and this might cause xss